packages/core/src/tools/limitModification.ts

```diff
@@ -54,5 +54,5 @@ function set(object: unknown, path: string, value: unknown) {
 }
 
 function isValidObjectContaining(object: unknown, field: string): object is { [key: string]: unknown } {
-  return typeof object === 'object' && object !== null && field in object
+  return typeof object === 'object' && object !== null && Object.prototype.hasOwnProperty.call(object, field)
```

Comment:
This change actually don't fix the issue due to https://github.com/github/codeql/issues/9869.
I'd be in favor of going with this fix and dismiss the alert as false positive.